lib/logdna/flush_scheduler.rb

```diff
@@ -0,0 +1,128 @@
+require 'securerandom'
+
+module FlushScheduler
```

Comment:
I feel like one of the primitives from concurrent-ruby would handle most of this complexity. `ScheduledTask` or `TimerTask` or something.

Reply:
I do see your comment about having more precise control, but I'm not sure if the added complexity is worth the trade off?  I don't have strong familiarity with the contract we want our flush scheduling to have, however.

Reply:
Totally glossed over that. After briefly reading it seems to make a lot of the same guarantees that this implementation is making with probably more reliability and features. I can probably replace this entire module with that.

Reply:
Switched over to `ScheduledTask`